docs/tutorial/Tutorial.md

```diff
@@ -0,0 +1,1064 @@
+This is the term that is traditionally used in the garbage collection literature 
+to describe the running program (because it 'mutates' the object graph).
+
+*plan*: A garbage collection algorithm expressed as a configuration of policies.
+
+See also [Plans and policies](#plans-and-policies) below.
+
+*policy*: A specific garbage collection algorithm, such as marksweep, copying, 
+immix, etc. Plans are made up of an arrangement of one or more policies. 
+
+See also [Plans and policies](#plans-and-policies) below.
+
+*scheduler*: Dynamically dispatches units of GC work to workers.
+
+*zeroing*, *zero initialization*: Initializing and resetting unused memory 
+bits to have a value of 0. Required by most memory-safe programming languages.
+
+See also: [Further Reading](#further-reading)
+
+
+#### Plans and Policies
+
+In MMTk, collectors are instantiated as plans, which can be thought of as 
+configurations of collector policies. In practice, most production 
+collectors and almost all collectors in MMTk are comprised of multiple 
+algorithms/policies. For example the gencopy plan describes a configuration 
+that combines a copying nursery with a semispace mature space. In MMTk we 
+think of these as three spaces, each of which happen to use the copyspace 
+policy, and which have a relationship which is defined by the gencopy plan. 
+Under the hood, gencopy builds upon a common plan which may also contain other 
+policies including a space for code, a read-only space, etc.
+
+Thus, someone wishing to construct a new collector based entirely on existing 
+policies may be able to do so in MMTk by simply writing a new plan, which is 
+what this tutorial covers.
+
+On the other hand, someone wishing to introduce an entirely new garbage 
+collection policy (such as Immix, for example), would need to first create 
+a policy which specifies that algorithm, before creating a plan which defines 
+how the GC algorithm fits together and utilizes that policy.
+
+
+[**Back to table of contents**](#contents)
+***
+## Preliminaries
+### Set up MMTk and OpenJDK
+#### Basic set up
+This tutorial can be completed with any binding. However, for the sake of 
+simplicity, only the setup for the OpenJDK binding will be described in detail 
+here. If you would like to use another binding, you will need to follow the 
+README files in their respective repositories 
+([JikesRVM](https://github.com/mmtk/mmtk-jikesrvm), 
+[V8](https://github.com/mmtk/mmtk-v8))
+ to set them up, and find appropriate benchmarks for testing. 
+ Also, while it may be useful to fork the relevant repositories to your own 
+ account, it is not required for this tutorial.
+
+First, set up OpenJDK, MMTk, and the binding:
+1. Clone the [OpenJDK binding](https://github.com/mmtk/mmtk-openjdk).
+2. Clone the mmtk-core repository and the 
+[OpenJDK VM repository](https://github.com/mmtk/openjdk). 
+Place them both in `mmtk-openjdk/repos`.
```

Comment:
Is this right???   mmtk-core should not be there, it should be in the parent director (ie a sibling, not child, of mmtk-openjdk).   Can someone check this please?

Reply:
You don't need to manually clone OpenJDK. You can just do `git submodule init` and `git submodule update`, which is instructed in the README for mmtk-openjdk. For mmtk-core, it can be put to any path - we just need to point to the path in `Cargo.toml`.

Reply:
At the moment, this is my file structure:
![image](https://user-images.githubusercontent.com/75106355/107467565-61925080-6bba-11eb-882e-0dc8c7511cba.png)

So it does work in the configuration I've got it in, but I can change the tutorial if it's not the proper file structure. I was a bit confused about it when I was starting up and haven't really thought about it since then, so I wouldn't be surprised if there was an issue.